evm-tests/ethcore-builtin/src/kzg.rs

```diff
@@ -0,0 +1,191 @@
+);
+
+/// Number of G1 Points.
+const NUM_G1_POINTS: usize = 4096;
+
+/// Number of G2 Points.
+const NUM_G2_POINTS: usize = 65;
+
+/// A newtype over list of G1 point from kzg trusted setup.
+#[derive(Debug, Clone, PartialEq, AsRef, AsMut, Deref, DerefMut)]
+#[repr(transparent)]
+struct G1Points(pub [[u8; BYTES_PER_G1_POINT]; NUM_G1_POINTS]);
+
+impl Default for G1Points {
+	fn default() -> Self {
+		Self([[0; BYTES_PER_G1_POINT]; NUM_G1_POINTS])
+	}
+}
+
+/// A newtype over list of G2 point from kzg trusted setup.
+#[derive(Debug, Clone, Eq, PartialEq, AsRef, AsMut, Deref, DerefMut)]
+#[repr(transparent)]
+struct G2Points(pub [[u8; BYTES_PER_G2_POINT]; NUM_G2_POINTS]);
+
+impl Default for G2Points {
+	fn default() -> Self {
+		Self([[0; BYTES_PER_G2_POINT]; NUM_G2_POINTS])
+	}
+}
+
+/// Default G1 points.
+const G1_POINTS: &G1Points = {
+	const BYTES: &[u8] = include_bytes!("../assets/g1_points.bin");
+	assert!(BYTES.len() == core::mem::size_of::<G1Points>());
+	unsafe { &*BYTES.as_ptr().cast::<G1Points>() }
+};
+
+/// Default G2 points.
+const G2_POINTS: &G2Points = {
+	const BYTES: &[u8] = include_bytes!("../assets/g2_points.bin");
+	assert!(BYTES.len() == core::mem::size_of::<G2Points>());
+	unsafe { &*BYTES.as_ptr().cast::<G2Points>() }
+};
+
+/// Parses the contents of a KZG trusted setup file into a list of G1 and G2 points.
+///
+/// These can then be used to create a KZG settings object with
+/// [`KzgSettings::load_trusted_setup`](c_kzg::KzgSettings::load_trusted_setup).
+#[allow(dead_code)]
+fn parse_kzg_trusted_setup(
+	trusted_setup: &str,
+) -> Result<(Box<G1Points>, Box<G2Points>), &'static str> {
+	let mut lines = trusted_setup.lines();
+
+	// load number of points
+	let n_g1 = lines
+		.next()
+		.ok_or("KzgFileFormatError")?
+		.parse::<usize>()
+		.map_err(|_| "KzgParseError")?;
+	let n_g2 = lines
+		.next()
+		.ok_or("KzgFileFormatError")?
+		.parse::<usize>()
+		.map_err(|_| "KzgParseError")?;
+
+	if n_g1 != NUM_G1_POINTS {
+		return Err("KzgMismatchedNumberOfPoints");
+	}
+
+	if n_g2 != NUM_G2_POINTS {
+		return Err("KzgMismatchedNumberOfPoints");
+	}
+
+	// load g1 points
+	let mut g1_points = Box::<G1Points>::default();
+	for bytes in &mut g1_points.0 {
+		let line = lines.next().ok_or("KzgFileFormatError")?;
+		hex::decode_to_slice(line, bytes).map_err(|_| "KzgParseError")?;
+	}
+
+	// load g2 points
+	let mut g2_points = Box::<G2Points>::default();
+	for bytes in &mut g2_points.0 {
+		let line = lines.next().ok_or("KzgFileFormatError")?;
+		hex::decode_to_slice(line, bytes).map_err(|_| "KzgParseError")?;
+	}
+
+	if lines.next().is_some() {
+		return Err("KzgFileFormatError");
+	}
+
+	Ok((g1_points, g2_points))
+}
+
+/// KZG Settings that allow us to specify a custom trusted setup.
+/// or use hardcoded default settings.
+#[allow(dead_code)]
+#[derive(Debug, Clone, Default)]
+pub enum EnvKzgSettings {
+	/// Default mainnet trusted setup
+	#[default]
+	Default,
+	/// Custom trusted setup.
+	Custom(Rc<KzgSettings>),
```

Comment:
I still put this part:

> Custom(Rc<KzgSettings>),

For investment tests fail for `EIP-4844`. If it will be after `EIP-4844` passed still unused it should be removed.